ocaml/xapi/xapi_vbd_helpers.ml

```diff
@@ -376,38 +376,17 @@ let clear_current_operations ~__context ~self =
 
 (** Check if the device string has the right form *)
 let valid_device dev ~_type =
-  let check_rest rest =
-    (* checks the rest of the device name = [] is ok, or a number is ok *)
-    if rest = [] then
-      true
-    else
-      try
-        ignore (int_of_string (String.implode rest)) ;
-        true
-      with _ -> false
-  in
   dev = "autodetect"
+  || Option.is_none (Device_number.of_string dev ~hvm:false)
   ||
-  match String.explode dev with
-  | 's' :: 'd' :: 'a' .. 'p' :: rest ->
-      check_rest rest
-  | 'x' :: 'v' :: 'd' :: 'a' .. 'p' :: rest ->
-      check_rest rest
-  | 'h' :: 'd' :: 'a' .. 'p' :: rest ->
-      check_rest rest
-  | 'f' :: 'd' :: 'a' .. 'b' :: rest ->
-      check_rest rest
-      (* QEMU only supports up to 2 floppy drives, hence fda or fdb *)
```

Comment:
This contradicts the limit in device_number, which allows for the staggering amount of 3 floppy drives

Reply:
qemu's manual agrees with the comment https://www.qemu.org/docs/master/system/qemu-manpage.html